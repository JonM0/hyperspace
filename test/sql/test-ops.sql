select '(1,1,1,1)'::fourd = '(1,1,1,1)'::fourd;
select '(1,1,1,1)'::fourd = '(2,2,2,2)'::fourd;

select '(1,1,1,1)'::fourd = '(+1,+1,+1,+1)'::fourd;

select '(1,1,1,1)'::fourd = '(2,1,1,1)'::fourd;
select '(1,1,1,1)'::fourd = '(1,2,1,1)'::fourd;
select '(1,1,1,1)'::fourd = '(1,1,2,1)'::fourd;
select '(1,1,1,1)'::fourd = '(1,1,1,2)'::fourd;

select '(1,1,1,1)'::fourd <> '(1,1,1,1)'::fourd;
select '(1,1,1,1)'::fourd <> '(2,2,2,2)'::fourd;

select '(1,1,1,1)'::fourd <> '(2,1,1,1)'::fourd;
select '(1,1,1,1)'::fourd <> '(1,2,1,1)'::fourd;
select '(1,1,1,1)'::fourd <> '(1,1,2,1)'::fourd;
select '(1,1,1,1)'::fourd <> '(1,1,1,2)'::fourd;


select '(1,2,3,4)'::fourd  + '(5,6,7,8)'::fourd = '(6,8,10,12)'::fourd;