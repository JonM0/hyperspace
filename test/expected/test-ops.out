select '(1,1,1,1)'::point4d = '(1,1,1,1)'::point4d;
 ?column? 
----------
 t
(1 row)

select '(1,1,1,1)'::point4d = '(2,2,2,2)'::point4d;
 ?column? 
----------
 f
(1 row)

select '(1,1,1,1)'::point4d = '(+1,+1,+1,+1)'::point4d;
 ?column? 
----------
 t
(1 row)

select '(1,1,1,1)'::point4d = '(2,1,1,1)'::point4d;
 ?column? 
----------
 f
(1 row)

select '(1,1,1,1)'::point4d = '(1,2,1,1)'::point4d;
 ?column? 
----------
 f
(1 row)

select '(1,1,1,1)'::point4d = '(1,1,2,1)'::point4d;
 ?column? 
----------
 f
(1 row)

select '(1,1,1,1)'::point4d = '(1,1,1,2)'::point4d;
 ?column? 
----------
 f
(1 row)

select '(1,1,1,1)'::point4d <> '(1,1,1,1)'::point4d;
 ?column? 
----------
 f
(1 row)

select '(1,1,1,1)'::point4d <> '(2,2,2,2)'::point4d;
 ?column? 
----------
 t
(1 row)

select '(1,1,1,1)'::point4d <> '(2,1,1,1)'::point4d;
 ?column? 
----------
 t
(1 row)

select '(1,1,1,1)'::point4d <> '(1,2,1,1)'::point4d;
 ?column? 
----------
 t
(1 row)

select '(1,1,1,1)'::point4d <> '(1,1,2,1)'::point4d;
 ?column? 
----------
 t
(1 row)

select '(1,1,1,1)'::point4d <> '(1,1,1,2)'::point4d;
 ?column? 
----------
 t
(1 row)

select '(1,2,3,4)'::point4d  + '(5,6,7,8)'::point4d = '(6,8,10,12)'::point4d;
 ?column? 
----------
 t
(1 row)

